atomic_reactor/util.py

```diff
@@ -1672,3 +1672,177 @@ def chain_get(d, path, default=None):
         except (IndexError, KeyError):
             return default
     return obj
+
+
+class DockerfileImages(object):
+    def __init__(self, dfp_parents):
+        self._original_parents = deepcopy(dfp_parents) if dfp_parents else []
+        self._custom_parent_image = False
+        self._custom_base_image = False
+        self._base_from_scratch = False
+        self._source_registry = None
+        self._organization = None
+        self._original_base_image = self._original_parents[-1] if self._original_parents else None
+        self._pullable_parents = []
+        self._local_parents = []
+        self._create_pullable()
+
+    @property
+    def original_parents(self):
+        """
+        provides unmodified parent_images from Dockerfile
+        """
+        return self._original_parents
+
+    @property
+    def original_base_image(self):
+        """
+        provides unmodified base image from Dockerfile
+        """
+        return self._original_base_image
+
+    @property
+    def base_image(self):
+        """
+        provides local tag for base_image, or base image with registry and enclosed
+        when we don't know local tag yet
+        """
+        if self.base_from_scratch:
+            return SCRATCH_FROM
+
+        if self._local_parents and self._local_parents[0]:
+            return deepcopy(self._local_parents[0])
+        elif self._pullable_parents and self._pullable_parents[0]:
+            return deepcopy(self._pullable_parents[0])
+        else:
+            raise KeyError('base_image')
+
+    @property
+    def base_image_key(self):
+        """
+        provides base image with registry and enclosed
+        (original base image with added registry and enclosed)
+        """
+        if self.base_from_scratch:
+            return SCRATCH_FROM
+
+        if self._pullable_parents and self._pullable_parents[0]:
+            return deepcopy(self._pullable_parents[0])
+        else:
+            raise KeyError('base_image')
```

Comment:
I'm not sure if KeyError is proper one. Maybe it should return `None` as unset property?

Reply:
well that is basically corner case for Dockerfile without any FROM
so calling 'base_image'/'base_image_key' is basically KeyError
I can change it to different error, but it should raise error for sure

Reply:
I didn't try to build with Dockerfile without FROM
this is really just corner case, also useful for unit tests

I don't have problem with RuntimeError

Reply:
the only thing is, that the class init gets list of parents (which are from Dockerfile, but in theory you can just use any list of parents) so that keyerror seems more appropriate

Reply:
also looking at code, it will actually fail already in set_df_path, with : Runtimerror when base_image is None